tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java

```diff
@@ -495,6 +495,48 @@ public static void test_throw_wrong_datatype() throws Exception {
 		conn.close();
 	}
 
+	public static void test_list_metadata() throws Exception {
+		try (Connection conn = DriverManager.getConnection("jdbc:duckdb:")) {
+			try (Statement stmt = conn.createStatement()) {
+				try (ResultSet rs = stmt.executeQuery("SELECT generate_series(2) as list")) {
+					ResultSetMetaData meta = rs.getMetaData();
+					assertEquals(meta.getColumnCount(), 1);
+					assertEquals(meta.getColumnName(1), "list");
+					assertEquals(meta.getColumnTypeName(1), "BIGINT[]");
+					assertEquals(meta.getColumnType(1), Types.ARRAY);
+				}
+			}
+		}
+	}
+
+	public static void test_struct_metadata() throws Exception {
+		try (Connection conn = DriverManager.getConnection("jdbc:duckdb:")) {
+			try (Statement stmt = conn.createStatement()) {
+				try (ResultSet rs = stmt.executeQuery("SELECT {'i': 42, 'j': 'a'} as struct")) {
+					ResultSetMetaData meta = rs.getMetaData();
+					assertEquals(meta.getColumnCount(), 1);
+					assertEquals(meta.getColumnName(1), "struct");
+					assertEquals(meta.getColumnTypeName(1), "STRUCT(i INTEGER, j VARCHAR)");
+					assertEquals(meta.getColumnType(1), Types.JAVA_OBJECT);
+				}
+			}
+		}
+	}
+
+	public static void test_map_metadata() throws Exception {
+		try (Connection conn = DriverManager.getConnection("jdbc:duckdb:")) {
```

Comment:
A bit of a nitpick, but you *can* collapse these three tries into one

Reply:
👍🏻